### ISSUE TITLE
constexpr static object gives an error but compiles

### ISSUE THREAD

**Issue by AMDG2** [microsoft/vscode-cpptools#1054]
I am using a [library](https://github.com/serge-sans-paille/frozen) to have constexpr unordered_map. When I a declare my map like this: 

```c++
void decode_1(const std::vector<uint8_t> & data) { }
void decode_2(const std::vector<uint8_t> & data) { }
void decode_3(const std::vector<uint8_t> & data) { }
void decode_4(const std::vector<uint8_t> & data) { }
void decode_5(const std::vector<uint8_t> & data) { }

typedef void (*decoder_t)(const std::vector<uint8_t> &);
constexpr static frozen::unordered_map<int, decoder_t, 5> decoders_map = {
    {1, decode_1},
    {2, decode_2},
    {3, decode_3},
    {4, decode_4},
    {5, decode_5}
};

int main(int argc, char ** argv)
{
    int id = atoi(argv[1]);
    auto it = decoders_map.find(id);
    if(it != decoders_map.end())
        std::cout << "Decoder not found for id: " << id << std::endl;
    else
        std::cout << "Decoder found for id: " << id << std::endl;
    return 0;
}
```

I get an error squiggle on the first `{`, the error is : 
```
expression must have a constant value -- access to uninitialized subobject (base class "std::__1::__tuple_leaf<1UL, std::__1::array<int64_t, 16UL>, false>")
std::initializer_list<std::__1::pair<int, decoder_t>>{&const std::__1::pair<int, decoder_t> [5]{std::__1::pair<int, decoder_t>{(int)1, decode_1}, std::__1::pair<int, decoder_t>{(int)2, decode_2}, std::__1::pair<int, decoder_t>{(int)3, decode_3}, std::__1::pair<int, decoder_t>{(int)4, decode_4}, std::__1::pair<int, decoder_t>{(int)5, decode_5}}, (size_t)5UL}
```

This code compiles fine with clang++ from Android toolchain (and should with g++ as stated by the library [README](https://github.com/serge-sans-paille/frozen#usage)).

I am using VSCode 1.15.1 (not insider) with CppTools extension version 0.12.4 on Linux (Ubuntu 10.04).

**Comment by sean-mcmanus** [comment]
Yeah, I also repro that using std::unordered_map. Actually, the bug repros on Windows too (previously I was hitting a crash that was preventing the error from appearing).

**Comment by sean-mcmanus** [comment]
This appears fixed in 0.17.7 -- not sure what release actually fixed it. 0.17.8-insiders has more fixes for other scenarios: https://github.com/Microsoft/vscode-cpptools/releases/tag/v0.17.8-insiders , and we made several more parser fixes that haven't been released yet.
